### ISSUE TITLE
Redshift percent_cont window function is unparsable

### ISSUE THREAD

**Issue by aviv** [sqlfluff/sqlfluff#2951]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

See the [relevant Redshift docs syntax page for the percentile_cont function](https://docs.aws.amazon.com/redshift/latest/dg/r_WF_PERCENTILE_CONT.html#r_WF_PERCENTILE_CONT-synopsis) 

This minimal valid Redshift query:

```sql
select
    dataset_id,
    (percentile_cont(0.20) within group (
        order by tract_percent_below_poverty asc
    ) over(partition by dataset_id)) as percentile_20
from dataset_with_census
```

Results in the following:
```
(venv3.9.5) ➜  sqlfluff git:(main) sqlfluff lint --dialect redshift /Users/aviv/Desktop/x.sql
== [/Users/aviv/Desktop/x.sql] FAIL                                   
L:   5 | P:   6 |  PRS | Line 5, Position 6: Found unparsable section: '
                       | over(partition by dataset_id)'
All Finished 📜 🎉!
```

That said, I modified the formatting of our original query a bit to make it play nicer with sqlfluff. When the query is formatted like this (still valid Redshift SQL):
```sql
select
    dataset_id,
    percentile_cont(0.40)
    within group (order by tract_percent_below_poverty asc)
    over(partition by dataset_id) as percentile_40
from dataset_with_census

```

The linter gets even more confused:
```
(venv3.9.5) ➜  sqlfluff git:(main) sqlfluff lint --dialect redshift /Users/aviv/Desktop/y.sql
== [/Users/aviv/Desktop/y.sql] FAIL                                   
L:   5 | P:   5 | L012 | Implicit/explicit aliasing of columns.       
L:   5 | P:   5 | L029 | Keywords should not be used as identifiers.
L:   5 | P:   9 |  PRS | Line 5, Position 9: Found unparsable section:
                       | '(partition by dataset_id) as percentile_...'
All Finished 📜 🎉!
```

### Expected Behaviour

I would expect both of the above queries to pass the linter since they are valid and correctly formatted SQL queries.

### Observed Behaviour

See description

### How to reproduce

See description

### Dialect

Redshift

### Version

```
(venv3.9.5) ➜  sqlfluff git:(main) sqlfluff --version
sqlfluff, version 0.11.1
```

### Configuration

I ran the above without any .sqluff or .sqlfluffignore config applied to simplify, but just for reference here is our config:

.sqlfluff:
```
[sqlfluff]
templater = dbt
exclude_rules = L031
dialect = redshift

[sqlfluff:rules]
tab_space_size = 2
indent_unit = space
max_line_length = 110

[sqlfluff:rules:L011]
# Aliasing preference for tables
aliasing = explicit

[sqlfluff:rules:L012]
# Aliasing preference for columns
aliasing = explicit

[sqlfluff:rules:L031]
disable = True
```

.sqlfluffignore:
```
target/
# dbt <1.0.0
dbt_modules/
# dbt >=1.0.0
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
The issue is due to this code in the Postgres dialect (that RedShift inherits from):

https://github.com/sqlfluff/sqlfluff/blob/42f1ca3a15f978a76f5c7ce082b1cb1776ee62ce/src/sqlfluff/dialects/dialect_postgres.py#L329-L334

This says that after the main function call you can have a `WithinGroupClauseSegment` OR a `OverClauseSegment` OR a `FilterClauseGrammar`.

Changing the `OneOf` to an `AnyNumberOf` allows both versions of you SQL to parse.

If you are indeed willing to submit a PR, then this is a pretty simple one to do. Just need to make that change and add a test case SQL, generate the YAML for it, and submit the PR. M<ore details here: https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#adding-test-cases-for-your-changes

**Comment by aviv** [comment]
Thanks so much @tunetheweb! I will take a shot at making that change and opening a PR.

**Comment by aviv** [comment]
Alright, PR is up: https://github.com/sqlfluff/sqlfluff/pull/2952 -- let me know if I missed anything! Thanks again for the quick response and for pointing out how easy this is to fix
